### PR TITLE
[CI] enable post-commit as a workflow action. 

### PR DESCRIPTION
**Pull request by cperkinsintel**
It'd be nice to be able to trigger post-commit checks, especially when we aren't able to test locally. This PR makes post-commit as a dispatchable workflow.  Liberte, Egalite, Fraternite

### CODE REVIEW

**Comment by Fznamznon** [comment]
Will it allow to run post commit jobs on a particular pull request?

**Comment by Fznamznon** [comment]
Could you please also revert https://github.com/intel/llvm/commit/5f649e21d8cd4645a44b4e7cbfee63fd969bc0ae , since you are touching sycl_post_commit.yml?
Eating machine time to revert dummy change doesn't seem reasonable to me.

**Comment by cperkinsintel** [comment]
@Fznamznon  - yes it will allow post-commit to be run on an individual PR.   Once this is merged, go to Actions and find post-commit.yml in the list on the left then click the button on the right to run it.

I don't see an extra line sycl_post_commit.yml at line 17. Was that commit merged?

File: .github/workflows/sycl_post_commit.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ on:
     - .github/workflows/sycl_linux_build_and_test.yml
     - .github/workflows/sycl_windows_build_and_test.yml
     - .github/workflows/sycl_macos_build_and_test.yml
+  workflow_dispatch:
```
Comment (pvchupin) [inline .github/workflows/sycl_post_commit.yml:17]:
```suggestion
```

**Review by pvchupin — COMMENTED** [review]
That should do the trick :)

**Review by pvchupin — APPROVED** [review]

**Comment by pvchupin** [comment]
We don't need to wait for actual result.

**Comment by cperkinsintel** [comment]
@Fznamznon  - the solution isn't perfect yet. Now that this is merged you can run the post-commit action on any branch. Of course, the branch for your PR is likely on your own fork, which doesn't have the CI resources available. In this case, push your branch to intel/llvm for the purposes of running the post-commit action on it, and then delete the branch after.

**Comment by Fznamznon** [comment]
> @Fznamznon - the solution isn't perfect yet. Now that this is merged you can run the post-commit action on any branch. Of course, the branch for your PR is likely on your own fork, which doesn't have the CI resources available. In this case, push your branch to intel/llvm for the purposes of running the post-commit action on it, and then delete the branch after.

I don't think pushing private branch to intel/llvm repo is a good thing. I believe we need a solution that doesn't require it.

**Pull request merged**